### PR TITLE
Marketplace tablet is a bit larger and lower DPI.

### PR DESCRIPTION
**Pull request by hyperlogic**
Also, includes bug fix for grabbing/equipping large objects, they no longer will drop immediately
if your grab point is too far away from the grabbed entity position.

### CODE REVIEW

**Comment by hyperlogic** [comment]
Testplan:
  * When using an HMD with hand controllers and you click on the marketplace icon.  A marketplace tablet should appear.  This tablet is slightly (0.1m) larger than master.  It is also 30dpi instead of 45dpi, making it easier to read.
  * It should be easier to grab the tablet in the corners, previously you could drop the tablet if too far away from the center.
  * Ensure the equip "tear-away" behavior still works.  Equip an object like the ping-pong ball.  With the opposite hand it should be possible to grab the object moving it away from your equip hand and unequip the object.  In fact the "tear-away" should be more predictable, as it is now based on the hand distance from the bounding box, not a simple position distance check.

**Comment by hifi-gustavo** [comment]
Build console output for commit: dcd91d43981a809c6fb3bd7d5ae96bc6404d0887 available at: https://gist.github.com/4c29265230d443ec4ca81f67c719b2a9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8537-dcd91d4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8537-dcd91d4.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0f42943bfbeb72088f6569e2e42e3315fe2fff7c available at: https://gist.github.com/4c29265230d443ec4ca81f67c719b2a9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8537-0f42943.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8537-0f42943.dmg

File: scripts/system/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -125,6 +125,12 @@ var ZERO_VEC = {
     z: 0
 };
 
+var ONE_VEC = {
```
Comment (jherico) [inline scripts/system/controllers/handControllerGrab.js]: there is an existing `Vec3.ONE` member that is this.

File: scripts/system/controllers/handControllerGrab.js [hunk 2]
```diff
@@ -232,6 +238,25 @@ CONTROLLER_STATE_MACHINE[STATE_ENTITY_TOUCHING] = {
...
 
+function distanceBetweenPointAndEntityBoundingBox(point, entityProps) {
+    var entityXform = new Xform(entityProps.rotation, entityProps.position);
+    var localPoint = entityXform.inv().xformPoint(point);
+    var minOffset = Vec3.multiplyVbyV(entityProps.registrationPoint, entityProps.dimensions);
+    var maxOffset = Vec3.multiplyVbyV(Vec3.subtract(ONE_VEC, entityProps.registrationPoint), entityProps.dimensions);
+    var localMin = Vec3.subtract(entityXform.trans, minOffset);
+    var localMax = Vec3.sum(entityXform.trans, maxOffset);
+
+    var v = {x: localPoint.x, y: localPoint.y, z: localPoint.z};
```
Comment (jherico) [inline scripts/system/controllers/handControllerGrab.js]: we should maybe add a  `Vec3.clamp(v, min, max)` function to do this.

File: scripts/system/controllers/handControllerGrab.js [hunk 3]
```diff
@@ -232,6 +238,25 @@ CONTROLLER_STATE_MACHINE[STATE_ENTITY_TOUCHING] = {
...
+    v.x = Math.min(v.x, localMax.x);
+    v.y = Math.max(v.y, localMin.y);
+    v.y = Math.min(v.y, localMax.y);
+    v.z = Math.max(v.z, localMin.z);
+    v.z = Math.min(v.z, localMax.z);
+
+    return Vec3.distance(v, localPoint);
+}
+
 function angleBetween(a, b) {
```
Comment (jherico) [inline scripts/system/controllers/handControllerGrab.js]: we should probably also have a `Vec3.angleBetween(a, b)`

**Comment by jherico** [comment]
My comments are non-blocking.  It would be nice to improve the `Vec3` interface, but not necessary.

**Comment by Nex-Pro** [comment]
Tested on Win10 with Vive.

Seems to work ok. Tablet is legible and usable.

Tear  away works fine.

Far grab works ok.

Notes:
Had a lot of strange laser issues like objects moving negative to the hand. Stray lasers. Not sure if this is due to this PR. But it is the first time I have encountered grabbed objects moving negative to the hand ( unintentionally). Also had the lasers grab point drop vertically on the object after grabbing.

**Comment by Nex-Pro** [comment]
I went back and forth between this and 5312 to compare the lasers. 5312 was 'normal'

**Pull request merged**